### PR TITLE
Calculate default mempool capacity as 2x the ledger max block size

### LINKED ISSUE #1467: Calculate the default mempool capacity based on the max block size from the ledger
Spoke with @dcoutts about this and he recommended that the default mempool capacity should be around 2-3x the max block size from the ledger. We should also allow this value to be manually overridden.

### PR DESCRIPTION
**Pull request by intricate**
Closes #1467

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs [hunk 1]
```diff
@@ -226,10 +236,13 @@ initInternalState NodeArgs { tracers, chainDB, registry, cfg,
                              initState, mempoolCap } = do
     varCandidates  <- newTVarM mempty
     varState       <- newTVarM initState
+    mpCap          <- atomically $ do
+      ledger <- ledgerState <$> ChainDB.getCurrentLedger chainDB
+      pure (mempoolCapacity ledger)
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: What if the max block size changes in the ledger? (Will this happen?) If that happens, we'll only change the capacity when restarting the node.

cc: @edsko @dcoutts

Comment (intricate) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: Should we support a dynamically changing mempool capacity? I wasn't sure if we wanted that additional complexity.

Comment (mrBliss) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: I would indeed avoid it. But basing its static size on a dynamically changing thing might be confusing (needs a comment). The main reason I'm commenting on this is to ask whether the max block size will actually change in practice.

Comment (dcoutts) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: The max block size will change. I presume the question is "if using the default size (no local override), is it acceptable to require a restart before picking up a change in the default block size?".

Consider what happens when syncing. We'll start with the size from the genesis and not change it. The mainnet genesis specifies a 2Mb block size, that was later reduced down to 64k (and will probably be bumped up again after the Shelley hard fork).

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: Seems to me that if operators override the max block size, they will want to keep an eye on the max block size on the chain anyway, and requiring a restart when this happens (which will be rare) doesn't seem like a huge deal.

**Pull request merged**